src/ORM/Entity.php

```diff
@@ -36,51 +36,77 @@ public function __construct(?array $data = null)
     /**
      * {@inheritdoc}
      *
-     * @throws \EoneoPay\Externals\ORM\Exceptions\InvalidMethodCallException If the method doesn't exist or is immutable
+     * Overrides and adds functionality to check for guarded properties in set calls.
      */
     public function __call(string $method, array $parameters)
     {
-        // Set available types
-        $types = ['get', 'has', 'is', 'set'];
+        // Look for a setter call
+        $matched = \preg_match('/^set([a-zA-Z][\w]+)$/i', $method, $matches);
```

Comment:
Can't this just search for `'/^set/` or substr?

Reply:
Could be, I adapted this from the previous __call code

Reply:
@edwardmurrell-loyaltycorp this is so the property name can be extracted and used on line 56